aws-codeguruprofiler-profilinggroup/src/main/java/software/amazon/codeguruprofiler/profilinggroup/UpdateHandler.java

```diff
@@ -47,18 +53,35 @@
             if (principals.isPresent()) {
                 putAgentPermissions(proxy, profilingGroupName, principals.get(), getPolicyResponse.revisionId());
                 logger.log(
-                    String.format("Policy for [%s] for accountId [%s] has been successfully updated! actionGroup: %s, principals: %s",
-                        profilingGroupName,
-                        awsAccountId,
-                        ActionGroup.AGENT_PERMISSIONS,
-                        principals.get())
+                        String.format("Policy for [%s] for accountId [%s] has been successfully updated! actionGroup: %s, principals: %s",
+                                profilingGroupName,
+                                awsAccountId,
+                                ActionGroup.AGENT_PERMISSIONS,
+                                principals.get())
                 );
             } else if (getPolicyResponse.policy() != null) {
                 removeAgentPermission(proxy, profilingGroupName, getPolicyResponse.revisionId());
                 logger.log(String.format("Policy for [%s] for accountId [%s] has been successfully removed!", profilingGroupName, awsAccountId));
             }
 
-            logger.log(String.format("%s [%s] for accountId [%s] has been successfully updated!", ResourceModel.TYPE_NAME, model.getProfilingGroupName(), awsAccountId));
+            // Current channels means the channels from the model
+            Optional<AnomalyDetectionNotificationConfiguration> anomalyDetectionNotificationConfiguration = anomalyDetectionNotificationConfiguration(model);
+            if (anomalyDetectionNotificationConfiguration.isPresent()) {
+                updateNotificationChannels(profilingGroupName, proxy,
+                        model.getAnomalyDetectionNotificationConfiguration().getChannels().stream().map(channel -> {
```

Comment:
There's a bit of a weird mixed usage here -- we use the `Optional` for the `isPresent`, but then access the model directly as well. I suggest not mixing both styles.